src/services/application.js

```diff
@@ -1,11 +1,25 @@
 const CompanyApplication = require("../models/CompanyApplication");
 const hash = require("../lib/passwordHashing");
+const ApplicationStatus = require("../models/constants/ApplicationStatus");
+const CompanyApplicationRules = require("../models/constants/CompanyApplicationRules");
+const NijobsBusinessRulesError = require("../lib/NijobsBusinessRulesError");
+
 
 class CompanyApplicationService {
 
     async create({
         email, password, companyName, motivation,
     }) {
+
+        const existingApplications = await CompanyApplication.find({ email });
+
+        if (existingApplications.some((application) =>
+            application.state === ApplicationStatus.PENDING ||
+            application.state === ApplicationStatus.APPROVED)
+        )  {
+            throw new NijobsBusinessRulesError(CompanyApplicationRules.ONLY_ONE_APPLICATION_ACTIVE_PER_EMAIL);
+        }
+
```

Comment:
Done, although I could not add a `beforeEach(clean accounts and company applications)` on the company application tests that check for email duplication, because this would run in parallel with the auth tests and would delete the accounts that were expected to exist there